externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/JdbcEventLogger.scala

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark.events
+
+import java.io._
+import java.sql.{PreparedStatement, SQLException, SQLIntegrityConstraintViolationException, Statement}
+import java.util.concurrent.TimeUnit
+
+import com.alibaba.druid.pool.DruidPooledConnection
+import com.google.common.cache.{Cache, CacheBuilder}
```

Comment:
Please ensure your are using the stable API of Guava, because the Kyuubi Spark SQL engine will pick the Guava from Spark runtime. Generally, it should be compatible with Guava 11, 14, 19, and not be marked @Beta after Guava 21.